packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -120,6 +145,237 @@ export default class VTable {
+
+    /**
+     * Format the border type
+     * @returns
+     */
+    private formatBorders(
+        format: TableFormat,
+        td: HTMLTableCellElement,
+        isFirstRow: boolean,
+        isLastRow: boolean,
+        isFirstColumn: boolean,
+        isLastColumn: boolean,
+        isLastColumnCell: boolean
+    ) {
+        this.setBorderColors(td, format);
+        switch (format.tableBorderFormat) {
+            case TableBorderFormat.DEFAULT:
+                return;
+            case TableBorderFormat.EXTERNAL:
+                if (!isFirstRow) {
+                    td.style.borderTopColor = TRANSPARENT;
+                }
+                if (!isLastRow) {
+                    td.style.borderBottomColor = TRANSPARENT;
+                }
+                if (!isFirstColumn) {
+                    td.style.borderLeftColor = TRANSPARENT;
+                }
+                if (!isLastColumn) {
+                    td.style.borderRightColor = TRANSPARENT;
+                }
+
+                break;
+            case TableBorderFormat.FIRST_COLUMN_HEADER_EXTERNAL:
+                if (!isFirstRow) {
+                    td.style.borderTopColor = TRANSPARENT;
+                }
+
+                if (!isLastRow && !isFirstRow) {
+                    td.style.borderBottomColor = TRANSPARENT;
+                }
+                if (!isFirstColumn) {
+                    td.style.borderLeftColor = TRANSPARENT;
+                }
+                if (!isLastColumn && !isFirstColumn) {
+                    td.style.borderRightColor = TRANSPARENT;
+                }
+                if (isFirstColumn && isFirstRow) {
+                    td.style.borderRightColor = TRANSPARENT;
+                }
+
+                break;
+            case TableBorderFormat.NO_HEADER_VERTICAL:
+                if (isFirstRow) {
+                    td.style.borderLeftColor = TRANSPARENT;
+                }
+                if (isLastColumnCell) {
+                    td.style.borderRightColor = TRANSPARENT;
+                }
+
+                break;
+            case TableBorderFormat.MIDDLE:
+                if (isFirstColumn) {
+                    td.style.borderLeftColor = TRANSPARENT;
+                }
+                if (isLastColumn) {
+                    td.style.borderRightColor = TRANSPARENT;
+                }
+        }
+    }
+
+    /**
+     * Organize the borders of table according to a border type
+     * @param format
+     * @returns
+     */
+
+    private setBordersType(format: TableFormat) {
+        this.cells.forEach((row, rowIndex) => {
+            row.forEach((cell, cellIndex) => {
+                if (cell.td) {
+                    this.formatBorders(
+                        format,
+                        cell.td,
+                        rowIndex === 0,
+                        rowIndex === this.cells.length - 1,
+                        cellIndex === 0,
+                        cellIndex === this.cells[0].length - 1,

```

Comment:
Although in most cases all rows have same count of cells, but let's do not assume that. You can check cellIndex == row.length - 1 here, same for next line